pkg/grpc/remote_authorizer.go

```diff
@@ -0,0 +1,171 @@
+package grpc
```

Comment:
I understand why you put this code here, though I would argue that `pkg/auth/remote_authorizer.go` would be a better spot for this. Or if this ends up introducing cycles, add a `pkg/auth/remote` or something.

The idea is that `pkg/auth` contains all authorization related code that can be used irrespective of the kind of transport that was used to make the incoming connection. In our case we can use RemoteAuthorizer both with incoming gRPC and HTTP requests.

What we have in `pkg/http` and `pkg/grpc` is the authentication related code, which is quite protocol specific (due to it receiving metadata differently and returning different styles of errors).

Reply:
I did have a cycle in the beginning of the development, but apparently not any more.